### DISCUSSION TITLE
Compatibility with Psiphon (Proxy)

### DISCUSSION THREAD

**Discussion by the-liquid-metal** [setvisible/DownZemAll#211] [Ideas]
ArrowDL works well if i don't set the proxy.

I use automatic proxy manager named [Psiphon](https://psiphon.ca/en/index.html) daily. This app assign random address to ```manual proxy setup``` section each time ```connect``` button clicked.

```settings``` > ```network & internet``` > ```proxy```
![image](https://github.com/user-attachments/assets/aa7dd55f-c051-4194-9a02-dd250803dd09)

```Psiphon window```
![image](https://github.com/user-attachments/assets/60f215e8-8b8c-4a66-bd5e-f43b82e3100a)

All my apps works well.

ArrowDL seems does not recognize this setting, and i have to adjust its own proxy setting each time Psiphon is used.

**Comment by setvisible** [comment]
Hi,

Thanks for reporting this bug.

The feature of proxy network exists in Qt6, so it was implemented following the Qt6 example. The implementation is done as best as possible, but at the moment there are no tests on it.

Let's keep this bug report opened for someone who would like to debug it.

Regards,
